src/test/java/org/wso2/carbon/apimgt/securityenforcer/PingAISecurityHandlerTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.apimgt.securityenforcer;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.wso2.carbon.apimgt.securityenforcer.dto.AISecurityHandlerConfig;
+import org.wso2.carbon.apimgt.securityenforcer.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.securityenforcer.utils.AISecurityException;
+import org.wso2.carbon.apimgt.securityenforcer.utils.AISecurityHandlerConstants;
+import org.wso2.carbon.apimgt.securityenforcer.utils.SecurityUtils;
+
+@RunWith(PowerMockRunner.class)
```

Comment:
We should avoid using PowerMock for unit testing in new code that we write. We use PowerMock to test interactions with legacy code that we cannot change. But if we need to use it for testing new code that we write it points to a design issue because the code is not testable.

It also maybe that the scenario that you are testing is more complex and therefore requires an integration test instead of a unit test.